### PR TITLE
Missing definition for `dns.resolveAny` and `dns.resolve(domain, 'ANY')` in `@types/node`

### PR DESCRIPTION
**Pull request by ZaneHannanAU**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

- [X] Provide a URL to documentation or source code which provides context for the suggested changes: <https://nodejs.org/docs/latest/api/dns.html#dns_dns_resolveany_hostname_callback> <https://nodejs.org/docs/latest/api/dns.html#dns_dns_resolve_hostname_rrtype_callback>

done late in the evening only trying to follow the style around the definitions.

Should probably be sorted at least.

### CODE REVIEW

**Review by BrunoScheufler — APPROVED** [review]

**Review by BrunoScheufler — APPROVED** [review]
LGTM 🎉

**Comment by ZaneHannanAU** [comment]
Cool.

**Review by islishude — CHANGES_REQUESTED** [review]
Please add tests for new adding.

**Comment by ZaneHannanAU** [comment]
On mobile ATM; I'll do what I can

```typescript
 dns.resolve("nodejs.org", "ANY", (err, addresses) => {
    const _addresses: ReadonlyArray<...Any*Record> = addresses;
});
dns.resolveAny("nodejs.org", (err, addresses) => {
    const _addresses: ReadonlyArray<...Any*Record> = addresses;
});
```
though the full record stack is giant.

Do you have a good way of saying one value is one of many types like the binary or in ts? The one that lets you `Array | Set` and so on?

Thank you.

**Comment by ZaneHannanAU** [comment]
I actually made a spelling mistake...

**Review by islishude — APPROVED** [review]

**Comment by ZaneHannanAU** [comment]
Thank you.

File: types/node/index.d.ts [hunk 1]
```diff
@@ -2607,6 +2634,7 @@ declare module "dns" {
     // NOTE: This namespace provides design-time support for util.promisify. Exported members do not exist at runtime.
     export namespace resolve {
         export function __promisify__(hostname: string, rrtype?: "A" | "AAAA" | "CNAME" | "NS" | "PTR"): Promise<string[]>;
```
Comment (sandersn) [inline types/node/index.d.ts]: This is not for this PR, but I noticed that `__promisify__(hostname: string, rrtype?: "A" | ...)` prevents `__promisify__(hostname: string, rrtype?: string)` from ever matching a call like `__promisify__("host")`. I'm pretty sure that `rrtype` should not be optional in the first overload.

**Review by sandersn — APPROVED** [review]

**Pull request merged**